### ISSUE TITLE
Improvement to math font selection

### ISSUE THREAD

**Issue by fred-wang** [MathML/MathMLinHTML5#26]
cc'ing @physikerwelt 

From http://www.mathml-association.org/MathMLinHTML5/output/S2.html#SS3.SSS2:

"Because mathematical formulas are generally written with special fonts, the default user agent stylesheet must reset the CSS font-family on the math element to serif. User agents should then use their own mechanism to try and interpret this serif value on the math element as a font with an OpenType MATH table."

1) This is maybe too specific to Gecko's implementation. The note should probably just suggest to make the user agent stylesheet change the font-family in order to force the selection of a math font.

2) To solve size inconsistency between the text font and math font, maybe introduce a new CSS keyword for font-size ("auto-adjust", "match-parent" or something like that) that would tell the CSS engines to automatically adjust the font-size of an element to take into account font-family change, so that the text height remains the same. See https://phabricator.wikimedia.org/T132607

**Comment by fred-wang** [comment]
There is also a proposal by @khaledhosny on mozilla's Bugzilla (https://bugzilla.mozilla.org/show_bug.cgi?id=947654#c1) which would not only solve size inconsistency but more generally find a math font that renders well with a text font. However, given that we currently only have a few fonts with a MATH table, I'm not sure that will work well at the moment.

**Comment by physikerwelt** [comment]
This is getting more and more complicated
https://de.wikipedia.org/wiki/MediaWiki_Diskussion:Common.css#Schriftgr.C3.B6.C3.9Fe_f.C3.BCr_MathML
While I understand German, it's hard to tell what exactly this conversation is about.

**Comment by physikerwelt** [comment]
@Edokter can you give a summary from your point of view?

**Comment by Edokter** [comment]
Complicated indeed... All I know is what is and what could be. 

Currently, the release version of Firefox still uses the following fonts list:
"Latin Modern Math, XITS Math, Cambria Math, TeX Gyre Bonum Math, TeX Gyre Pagella Math, TeX Gyre Schola, TeX Gyre Termes Math, STIX Math, Asana Math, STIXGeneral, DejaVu Serif, DejaVu Sans, Symbol, Times New Roman"

Mozilla is thinking about changing it to just "serif", and let Gecko look up the most suitable font. I don't know what the ramifications will be: I only have Windows to test with. I also don't know how "consistent" is defined; consistent with running text (parent font), or between platforms? both seem impossible, as not many mainstream fonts have have any extensive math support to speak of. That is not even considering webfonts becoming more common.

What I see for MathML is probably a platform-dependent font stack, which picks the most suitable font per platform. On Windows, Cambria would be the most obvious choice, as it comes with a companion math font (replacing the older Times New Roman/MT Extra combination).

**Comment by fred-wang** [comment]
> Mozilla is thinking about changing it to just "serif", and let Gecko look up the most suitable font. I

No, bug 947654 is fixed since Gecko 41. Gecko uses "serif" in its user stylesheet and the actual selection is done via the x-math property. This property can be edited using the normal font preference menu.

> I also don't know how "consistent" is defined; consistent with running text (parent font), or between platforms? 

As I said it is consistency *between the text font and math font*. Yes, that's impossible right now because of the few math fonts and no standard CSS way to do that. This issue is for a long term discussion for Web engines and OS developers (not only Firefox's way). Unfortunately, it's won't solve the problem in MediaWiki in the short term so that's why it's best to use the SVG fallback by default and let advanced users configure the MathML style themselves.
